sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -53,6 +53,16 @@ def __init__(self, hidden_layer_sizes, activation, solver,
                  warm_start, momentum, nesterovs_momentum, early_stopping,
                  validation_fraction, beta_1, beta_2, epsilon):
         self.activation = activation
+        # If a single activation function is given,
+        # use it for each hidden layer
+        if type(self.activation) == str and\
+                hasattr(hidden_layer_sizes, "__iter__"):
+            self.activation_fncs = [self.activation] * len(hidden_layer_sizes)
+        elif type(self.activation) == str and\
+                not hasattr(hidden_layer_sizes, "__iter__"):
```

Comment:
I assume that you want to check that this is an iterable. I would use

```python
import collections

isinstance(hidden_layer_sizes, collections.Iterable)
```

Reply:
done!